### PR TITLE
IFATQCZZ-124 - Finished DSL Layer

### PR DESCRIPTION
**Pull request by rkliuha**

### CODE REVIEW

File: src/test/java/academy/softserve/edu/pageobjects/AddItemPage.java [hunk 1]
```diff
@@ -37,46 +37,53 @@ public AddItemPage(final WebDriver driver) {
...
 
-    public final AddItemPage selectItemDimensionDropdown(final String itemDimension) {
+    public final AddItemPage selectItemDimension(final String itemDimension) {
 
         itemDimensionDropdown.sendKeys(itemDimension);
         return this;
     }
 
-    public final AddItemPage selectSearchForItemDropdown(final String condition) {
+    public final AddItemPage selectSearchForItem(final String condition) {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/AddItemPage.java]: setSearchCondition() ?

File: src/test/java/academy/softserve/edu/pageobjects/AddProductPage.java [hunk 2]
```diff
@@ -36,34 +36,51 @@ public AddProductPage(final WebDriver driver) {
...
+                .fillProductDescription(product.getProductDescription())
+                .fillProductPrice(String.valueOf(product.getProductPrice()));
+    }
+
+    public final ItemManagementPage createProduct() {
+
+        return clickOk();
+    }
+
+    public final ItemManagementPage doNotCreateProduct() {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/AddProductPage.java]: cancelCreatingProduct()

File: src/test/java/academy/softserve/edu/pageobjects/AdministrationPage.java [hunk 3]
```diff
@@ -93,124 +93,128 @@ public AdministrationPage(final WebDriver driver) {
...
 
-    public final AdministrationPage clickShowQuantityOfItems() {
+    public final AdministrationPage showQuantityOfItems() {
 
         showQuantityOfItems.click();
         return this;
     }
 
-    public final AdministrationPage clickFirstNameHeaderButton() {
+    public final AdministrationPage sortTableByFirstName() {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/AdministrationPage.java]: can we combine sort methods into something like sortTableBy(Enum.FIRSTNAME) ?

File: src/test/java/academy/softserve/edu/pageobjects/CreateNewOrderPage.java [hunk 4]
```diff
@@ -145,117 +146,123 @@ public CreateNewOrderPage(final WebDriver driver) {
         super(driver);
     }
 
-    public final OrderItemsErrorMessagePage clickSaveButtonFail() {
+    public final OrderItemsErrorMessagePage saveOrderWithFail() {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/CreateNewOrderPage.java:150]: add TODO

File: src/test/java/academy/softserve/edu/pageobjects/CreateNewOrderPage.java [hunk 5]
```diff
@@ -145,117 +146,123 @@ public CreateNewOrderPage(final WebDriver driver) {
...
-    public final CreateNewOrderPage clickFirstItemDeleteButton() {
+    public final CreateNewOrderPage choosePreferableDeliveryDate() {
 
-        firstItemDeleteButton.click();
+        preferableDeliveryDateChooseLink.click();
         return this;
     }
 
-    public final CreateNewOrderPage clickCalendarTuesdayElement() {
+    public final CustomerOrderingPage doOrder() {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/CreateNewOrderPage.java]: rename createOrder()

File: src/test/java/academy/softserve/edu/pageobjects/CreateNewUserPage.java [hunk 6]
```diff
@@ -42,55 +43,67 @@ public CreateNewUserPage(WebDriver driver) {
 
     private final Button cancelButton = new Button(driver, CANCEL_BUTTON);
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/CreateNewUserPage.java:47]: consider renaming fill\* methods to set\* methods

File: src/test/java/academy/softserve/edu/pageobjects/CustomerOrderingPage.java [hunk 7]
```diff
@@ -48,52 +48,59 @@ public CustomerOrderingPage(WebDriver driver) {
...
 
-    public final CreateNewOrderPage clickCreateNewOrderLink() {
+    public final CreateNewOrderPage goToCreateNewOrderPage() {
 
         createNewOrderLink.click();
         return new CreateNewOrderPage(driver);
     }
 
-    public final CustomerOrderingPage clickResizeShowItemsLink() {
+    public final CustomerOrderingPage resizeShowOrders() {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/CustomerOrderingPage.java]: changeItemsPerPage()

File: src/test/java/academy/softserve/edu/pageobjects/CustomerOrderingPage.java [hunk 8]
```diff
@@ -48,52 +48,59 @@ public CustomerOrderingPage(WebDriver driver) {
...
     }
 
-    public final CustomerOrderingPage fillSearchInput(final String searchText) {
+    public final CustomerOrderingPage fillSearch(final String searchText) {
 
         searchInput.sendKeys(searchText);
         return this;
     }
 
+    public final CustomerOrderingPage searchOrder(final String searchCondition, final String searchText) {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/CustomerOrderingPage.java]: rename searchForOrder(), rename all similar

File: src/test/java/academy/softserve/edu/pageobjects/EditProductPage.java [hunk 9]
```diff
@@ -27,34 +28,52 @@ public EditProductPage(final WebDriver driver) {
...
+                .fillProductPrice(String.valueOf(product.getProductPrice()));
+
+    }
+
+    public final ItemManagementPage editProduct() {
+
+        return clickOk();
+    }
+
+    public final ItemManagementPage doNotEditProduct() {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/EditProductPage.java]: cancelEditingProduct()

File: src/test/java/academy/softserve/edu/pageobjects/EditUserPage.java [hunk 10]
```diff
@@ -40,45 +41,79 @@ public EditUserPage(final WebDriver driver) {
...
 
-    public final EditUserPage fillEmailAddressInput(final String emailAddress) {
+    public final EditUserPage fillEmailAddress(final String emailAddress) {
 
         emailAddressInput.sendKeys(emailAddress);
         return this;
     }
 
-    public final EditUserPage selectRegionDropdown(final String region) {
+    public final EditUserPage selectRegion(final String region) {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/EditUserPage.java]: can we use enum for that?

File: src/test/java/academy/softserve/edu/pageobjects/EditUserPage.java [hunk 11]
```diff
@@ -40,45 +41,79 @@ public EditUserPage(final WebDriver driver) {
...
 
         regionDropdown.sendKeys(region);
         return this;
     }
 
-    public final AdministrationPage clickSaveChangesButton() {
+    public final EditUserPage selectRole(final String role) {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/EditUserPage.java]: can we use enum for that?

File: src/test/java/academy/softserve/edu/pageobjects/EditUserPage.java [hunk 12]
```diff
@@ -40,45 +41,79 @@ public EditUserPage(final WebDriver driver) {
...
+                .selectRegion(newUser.getRegionName())
+                .selectRole(newUser.getRoleName());
+    }
+
+    public final AdministrationPage editUser() {
+
+        return clickSaveChanges();
+    }
+
+    public final AdministrationPage doNotEditUser() {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/EditUserPage.java]: cancelEditingUser()

File: src/test/java/academy/softserve/edu/pageobjects/ItemManagementPage.java [hunk 13]
```diff
@@ -86,100 +86,107 @@ public ItemManagementPage(final WebDriver driver) {
...
 
-    public final ItemManagementPage clickDeleteProductLinkById(final int productId) {
+    public final ItemManagementPage deleteProductById(final int productId) {
 
         getDeleteProductLinkById(productId).click();
         return this;
     }
 
-    public final ItemManagementPage clickResizeProductsListButton() {
+    public final ItemManagementPage resizeShowProducts() {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/ItemManagementPage.java]: done, see previous

File: src/test/java/academy/softserve/edu/pageobjects/ItemManagementPage.java [hunk 14]
```diff
@@ -86,100 +86,107 @@ public ItemManagementPage(final WebDriver driver) {
...
 
         resizeProductsListButton.click();
         return this;
     }
 
-    public final ItemManagementPage clickNameHeaderButton() {
+    public final ItemManagementPage sortTableByName() {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/ItemManagementPage.java]: can we combine with enum use?

File: src/test/java/academy/softserve/edu/pageobjects/MerchandiserOrderingPage.java [hunk 15]
```diff
@@ -79,100 +79,107 @@ public MerchandiserOrderingPage(final WebDriver driver) {
...
 
-    public final MerchandiserOrderingPage clickApplyButton() {
+    public final MerchandiserOrderingPage clickApply() {
 
         applyButton.click();
         return this;
     }
 
-    public final MerchandiserOrderingPage clickOrderNameHeaderButton() {
+    public final MerchandiserOrderingPage sortTableByOrderName() {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/MerchandiserOrderingPage.java]: can we combine sort methods with enum use?

File: src/test/java/academy/softserve/edu/pageobjects/UserInfoPage.java [hunk 16]
```diff
@@ -48,37 +48,37 @@ public UserInfoPage(final WebDriver driver) {
...
 
-    public final MerchandiserOrderingPage clickMerchandiserOrderingTab() {
+    public final MerchandiserOrderingPage goToMerchandiserOrderingPage() {
 
         merchandiserOrderingLink.click();
         return new MerchandiserOrderingPage(driver);
     }
 
-    public UserInfoPage clickUkrainianButton() {
+    public UserInfoPage changeLanguageToUkrainian() {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/pageobjects/UserInfoPage.java]: redo into using enum - changeLanguageTo(Enum.UA);

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/academy/softserve/edu/pageobjects/AddProductPage.java (hunk 2 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/AddProductPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/CreateNewOrderPage.java (hunk 4 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/CreateNewOrderPage.java:150]: Done

File: src/test/java/academy/softserve/edu/pageobjects/CreateNewOrderPage.java (hunk 5 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/CreateNewOrderPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/CustomerOrderingPage.java (hunk 7 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/CustomerOrderingPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/CustomerOrderingPage.java (hunk 8 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/CustomerOrderingPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/EditProductPage.java (hunk 9 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/EditProductPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/EditUserPage.java (hunk 12 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/EditUserPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/ItemManagementPage.java (hunk 13 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/ItemManagementPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/UserInfoPage.java (hunk 16 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/UserInfoPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/EditUserPage.java (hunk 10 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/EditUserPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/EditUserPage.java (hunk 11 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/EditUserPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/AdministrationPage.java (hunk 3 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/AdministrationPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/ItemManagementPage.java (hunk 14 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/ItemManagementPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/MerchandiserOrderingPage.java (hunk 15 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/MerchandiserOrderingPage.java]: Done

File: src/test/java/academy/softserve/edu/pageobjects/CreateNewUserPage.java (hunk 6 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/CreateNewUserPage.java:47]: Done

File: src/test/java/academy/softserve/edu/pageobjects/AddItemPage.java (hunk 1 above)
Comment (rkliuha) [reply src/test/java/academy/softserve/edu/pageobjects/AddItemPage.java]: Done

**Pull request merged**